chromium/background.js

```diff
@@ -497,11 +497,39 @@ function onErrorOccurred(details) {
   }
 }
 
+/**
+ * handle webrequest.onHeadersReceived, insert upgrade-insecure-requests directive
+ * @param details details for the chrome.webRequest (see chrome doc)
+ */
+function onHeadersReceived(details) {
+  if (isExtensionEnabled && httpNowhereOn) {
```

Comment:
There may be a wild edge case here.  In https://github.com/EFForg/https-everywhere/blob/28087477257523c65d14bd5b4e1e79ad84056c08/chromium/background.js#L247-L253, we specifically allow `.onion` URLs that do not have HTTPS in HTTP Nowhere mode.  If a `.onion` URL serves only one page in HTTPS, and the rest are over HTTP, landing on that page with this logic will cause the client to make unreachable upgraded requests over HTTPS.

As I said, it's a pretty far-fetched edge case (why would a `.onion` site have a TLS configuration like that?) but it is possible, and probably worth accounting for.

Reply:
I agree. Since `.onion` requests are encrypted anyway, IMHO we shall exclude them from the HTTPS upgrades.

Reply:
I think requests to non-`.onion` websites should still be upgraded if they were made from `.onion` website.